src/ConfigurationFactory.php

```diff
@@ -113,7 +113,7 @@ public static function init(
 
         //Validate settings structure if we have validation and key is default or test
         if (!empty($validator) && !empty($constraints) && ($key === self::DEFAULT_KEY || $key === self::ENV_TEST)) {
-            $violations = $validator->validate($settings, $constraints);
+            $violations = $validator->validate(array_merge($settings, $default), $constraints);
```

Comment:
This is so that validation works if you keep stuff in common. Otherwise common can not contain anything required.